build.gradle

```diff
@@ -849,7 +849,7 @@ project(':core') {
     compile.exclude module: 'mail'
     // To prevent a UniqueResourceException due the same resource existing in both
     // org.apache.directory.api/api-all and org.apache.directory.api/api-ldap-schema-data
-    testCompile.exclude module: 'api-ldap-schema-data'
+    all*.exclude module: 'api-ldap-schema-data'
```

Comment:
Is it related to the fact that we changed a bunch of things from `compile` to `implementation`/`api` and from `testCompile` to `testImplementation`? Does it also cause problems for the `compile` lines above the one you changed?

Reply:
> Does it also cause problems for the `compile` lines above the one you changed?

No, not at all. It only modifies exclusion level for a single dependency (referenced here as a module 'api-ldap-schema-data'). Isolated change, atomic level.

As for a your first question:
> Is it related to the fact that we changed a bunch of things from `compile` to `implementation`/`api` and from `testCompile` to `testImplementation`?

 you just helped me to realize that this was the **last** `testCompile` reference in a project !

Some details: 
* Gradle 6.8.3: `testCompile` is deprecated and superseded by `testImplementation`: https://docs.gradle.org/6.8.3/userguide/java_plugin.html#sec:java_plugin_and_dependency_management
* Gradle 7.0.1: `testCompile` is removed (doesn't exists): https://docs.gradle.org/7.0.1/userguide/java_plugin.html#sec:java_plugin_and_dependency_management

So, puzzle is solved, I guess: I assume that Gradle 7 is just skipping  directive `testCompile.exclude module: 'api-ldap-schema-data`  and hence 'UniqueResourceException' are rising again from the dust... 
Now, **this** could be a regression/bug in Gradle 7, I guess I will inform Gradle team.

And of course, now I do realize that we could try to switch from `testCompile` to `testImplementation' for this exclusion also (it is looks appropriate). I will test this on my end.

Reply:
> No, not at all. It only modifies exclusion level for a single dependency (referenced here as a module 'api-ldap-schema-data'). Isolated change, atomic level.

You misunderstood my point. `compile` was also removed, so those lines may not be having any impact.

Reply:
> You misunderstood my point. compile was also removed, so those lines may not be having any impact.

Aha, I see now.... I was focused on `testCompile` only... You have a point, `compile` is removed in Gradle 7 also and given a fact that no error is thrown I assume that Gradle 7 just skips in silence (at the surface, at least) this entire block of exclusions: https://github.com/apache/kafka/blob/ebb1d6e21cc9213071ee1c6a15ec3411fc215b81/build.gradle#L818

Well, this could really be a Gradle regression; I filed a ticket here: https://github.com/gradle/gradle/issues/17187 so I will just update it.

Reply:
Let me think what would be a good configurations to replace `compile` and `testCompile` (I think that suggested way for exclusion is to use resolvable  https://docs.gradle.org/6.8.3/userguide/java_library_plugin.html#sec:java_library_separation but I have to double-check)